### PR TITLE
Remove elastic index registry logic

### PR DESCRIPTION
**Pull request by joxl**
## Technical Summary

This registry module is no longer needed (has been obsolete ever since the document adapters were implemented). This PR is the predecessor to another (forthcoming) that will implement migration utilities for index management.

This change has added value now that both @AmitPhulera and I are both working on the Elasticsearch upgrade task because it reduces extra complexity for his work.

One huge "byproduct" gain of this change is that the `es_test` decorator now implements the Elastic index setup/teardown logic which I've been meaning to add for a long time. There is a lot of discrete Elastic index setup/teardown logic in tests that can now be ripped out when we have the time for that cleanup.

Review commit by commit.

Jira: [SAAS-14003](https://dimagi-dev.atlassian.net/browse/SAAS-14003)

### Safety story

Change is fully covered by tests. Majority of changes _are_ test code. Production code changes are safe because their function is already superseded by the adapter layer.

### Automated test coverage

Covered by automated tests.

### QA Plan

Invisible, no QA.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/tests/test_bulk.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def test_get_errors_with_ids(self):
 
 
 @sharded
-@es_test(index=TEST_INDEX_INFO)
+@es_test
```
Comment (AmitPhulera) [inline corehq/ex-submodules/pillowtop/tests/test_bulk.py:55]: What does this decorator do in these tests?

**Review by snopoke — APPROVED** [review]
Nice cleanup. A few small comments but nothing blocking.

File: corehq/ex-submodules/pillowtop/tests/test_bulk.py (hunk 1 above)
Comment (joxl) [reply corehq/ex-submodules/pillowtop/tests/test_bulk.py:55]: It adds the `es_test` nose attribute, so you can run all decorated tests like this:

```shell
./manage.py test --attr=es_test
```

Comment (AmitPhulera) [reply corehq/ex-submodules/pillowtop/tests/test_bulk.py:55]: Ohh! Got it. Thanks 👍

**Review by AmitPhulera — APPROVED** [review]

File: corehq/apps/es/tests/test_test_utils.py [hunk 2]
```diff
@@ -1,180 +1,153 @@
...
+    try:
+        @es_test(requires=[pigs_adapter], setup_class=True)
+        class TestExpectsClassmethod:
 
             def setUpClass(self):
                 pass
 
+        assert False, "ValueError not raised"
+    except ValueError as exc:
+        assert str(exc).startswith("'setup_class' expects a classmethod"), str(exc)
```
[the remark this answers is not in the document]
Comment (joxl) [reply corehq/apps/es/tests/test_test_utils.py]: See 21d93912

**Review by AmitPhulera — APPROVED** [review]
⭐

**Comment by AmitPhulera** [comment]
I just merged the reindex PR which resulted in some conflicts with this branch.

**Review by AmitPhulera — APPROVED** [review]

**Pull request merged**